### ISSUE TITLE
NgbTimepicker does not display a valid value on the second setting of an invalid value

### ISSUE THREAD

**Issue by Vlatislav** [ng-bootstrap/ng-bootstrap#4477]
### Bug description:
ngb-timepicker:
Set this control in a angular form
Fill in the timepicker, for example, with the data 01:01. 
Next are the steps to reproduce the bug. 
For example, enter the number 33 in the hour field and click outside of the timepicker. The value in the field will change to 09h - a valid value. Next, enter the number 33 again in the same hour field and click outside of this control. As a result, the value in the control will be a valid value with 09 hours and minutes. But 33 hours are displayed in the hour field of timepicker.

### Link to minimally-working StackBlitz that reproduces the issue:

Goto https://ng-bootstrap.github.io/#/components/timepicker/examples, inspect the control to see the classes set on the control.

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 9 and 15

ng-bootstrap: 6.1.0 and 14.1.0

Bootstrap: 4.5.0

**Comment by Vlatislav** [comment]
the same thing happens with the minutes field.
the interest is that if we do not insert the same wrong values in a row, the error will not be reproduced.

**Comment by RAJEEV2510** [comment]
Yeah It Happens When Will you insert the wrong value again its happen

**Comment by vairavlavy** [comment]
Please fix this ASAP

**Comment by RAJEEV2510** [comment]
I will fix this soon ,hope they will merge my pr.
